docs/root/version_history/current.rst

```diff
@@ -46,6 +46,14 @@ Changes
   tracing is not forced.
 * router: allow retries of streaming or incomplete requests. This removes stat `rq_retry_skipped_request_not_complete`.
 * router: allow retries by default when upstream responds with :ref:`x-envoy-overloaded <config_http_filters_router_x-envoy-overloaded_set>`.
+* router: internal redirect configs are moved to the :ref`internal_redirect_policy
+  <envoy_api_field_router.RouterAction.internal_redirect_policy>` field, which defines more fine
+  grained control of the internal redirect behavior. This changes the default cross scheme redirect behavior.
+  All cross scheme redirect is allowed by default, but http downstream to https target is disallowed. To restore
```

Comment:
I think you also need to doc up the deprecated field in the Deprecated section and maybe move comments on how to preserve behavior there?

Reply:
Done.